### DISCUSSION TITLE
Help configuring Guidance with Azure

### DISCUSSION THREAD

**Discussion by doomgrave** [guidance-ai/guidance#553] [Q&A] [answered]
Hi all.
I cannot run guidance on Azure Openai, may you suggest me whats the problem?
I've uninstalled and reinstalled guidance, tiktoken, openai modules, but still i get the error below.

PS: the data shared are scrubbled as an example.

CODE:
azure_api_key = 'api_key_here'
azure_version='2023-07-01-preview'
azure_model = 'gpt-4'
azure_deployment_url='https://mydomain.openai.azure.com/openai/deployments/AITEST/chat/completions?api-version=2023-07-01-preview'

azureLLM = guidance.models.AzureOpenAI(azure_endpoint=azure_deployment_url,api_key=azure_api_key,version=azure_version,model=azure_model)

ERROR:
Exception has occurred: AttributeError
module 'guidance.models' has no attribute 'AzureOpenAI'
  File "C:\Guidance_Examples\Guidance_Llama_llamacpp_Azure.py", line 26, in <module>
azureLLM = guidance.models.AzureOpenAI(
azure_endpoint=azure_deployment_url, 
api_key=azure_api_key,
version=azure_version,
model=azure_model
)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'guidance.models' has no attribute 'AzureOpenAI'

**Comment by Harsha-Nori** [comment]
Hi @doomgrave, what version of the guidance library are you running? You can do:

```python
import guidance
print(guidance.__version__)
```

 to check.

**Comment by doomgrave** [comment, marked as answer]
PROBLEM SOLVED!
I was running the test from the Cloned repository, so even if my version of guidance was updated it was reading the local ./guidance folder with an older version!
Really stupid but i was going crazy!

**Reply by Harsha-Nori** [reply]
Glad it all worked out 🙂
